aioamqp/protocol.py

```diff
@@ -327,7 +333,7 @@ def send_heartbeat(self):
         """
         frame = amqp_frame.AmqpRequest(self._stream_writer, amqp_constants.TYPE_HEARTBEAT, 0)
         request = amqp_frame.AmqpEncoder()
-        frame.write_frame(request)
+        yield from self._write_frame(frame, request)
```

Comment:
so calling `yield from` from a _function_ does _NOT_ work…

Back to the drawing board.